### PR TITLE
chore: adding Dockerfile and make commands for building/running

### PR DESCRIPTION
**Pull request by amanda103**
https://unstructured-ai.atlassian.net/browse/CORE-742

Steps to run:

```
# this step will take a while:
(unstructured) ➜  unstructured git:(amanda/add-Dockerfile) make docker-build

(unstructured) ➜  unstructured git:(amanda/add-Dockerfile) make docker-start-bash
docker run --platform linux/amd64 -ti --rm unstructured-dev:latest
[notebook-user@6cf610cd9177 ~]$ python3
Python 3.8.15 (default, Mar 10 2023, 21:05:24)
[GCC 7.3.1 20180303 (Red Hat 7.3.1-5)] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> from unstructured.partition.text import partition_text
[nltk_data] Downloading package punkt to /home/notebook-
[nltk_data]     user/nltk_data...
[nltk_data]   Unzipping tokenizers/punkt.zip.
[nltk_data] Downloading package averaged_perceptron_tagger to
[nltk_data]     /home/notebook-user/nltk_data...
[nltk_data]   Unzipping taggers/averaged_perceptron_tagger.zip.
>>> elements = partition_text(filename="example-docs/fake-text.txt")
>>> [str(e) for e in elements]
['This is a test document to use for unit tests.', 'Doylestown, PA 18901', 'Important points:', 'Hamburgers are delicious', 'Dogs are the best', 'I love fuzzy blankets']
```

### CODE REVIEW

**Comment by cragwolfe** [comment]
Looks good! Can you document what additional dependencies are needed to support all file types? E.g., libreoffice for .doc, etc.

**Comment by ryannikolaidis** [comment]
lgtm! Built and ran the manual test and all was successful.

I did notice the logging error at the end of the build (did others hit that? assuming known and innocuous?)

```#22 1140.0 --- Logging error ---
#22 1140.0 Traceback (most recent call last):
#22 1140.0   File "/home/notebook-user/.local/lib/python3.8/site-packages/pip/_internal/utils/logging.py", line 177, in emit
#22 1140.0     self.console.print(renderable, overflow="ignore", crop=False, style=style)
#22 1140.0   File "/home/notebook-user/.local/lib/python3.8/site-packages/pip/_vendor/rich/console.py", line 1673, in print
#22 1140.0     extend(render(renderable, render_options))
#22 1140.0   File "/home/notebook-user/.local/lib/python3.8/site-packages/pip/_vendor/rich/console.py", line 1305, in render
#22 1140.0     for render_output in iter_render:
#22 1140.0   File "/home/notebook-user/.local/lib/python3.8/site-packages/pip/_internal/utils/logging.py", line 134, in rich_console
#22 1140.0     for line in lines:
#22 1140.0   File "/home/notebook-user/.local/lib/python3.8/site-packages/pip/_vendor/rich/segment.py", line 249, in split_lines
#22 1140.0     for segment in segments:
#22 1140.0   File "/home/notebook-user/.local/lib/python3.8/site-packages/pip/_vendor/rich/console.py", line 1283, in render
#22 1140.0     renderable = rich_cast(renderable)
#22 1140.0   File "/home/notebook-user/.local/lib/python3.8/site-packages/pip/_vendor/rich/protocol.py", line 36, in rich_cast
#22 1140.0     renderable = cast_method()
#22 1140.0   File "/home/notebook-user/.local/lib/python3.8/site-packages/pip/_internal/self_outdated_check.py", line 130, in rich
#22 1140.0     pip_cmd = get_best_invocation_for_this_pip()
#22 1140.0   File "/home/notebook-user/.local/lib/python3.8/site-packages/pip/_internal/utils/entrypoints.py", line 58, in get_best_invocation_for_this_pip
#22 1140.0     if found_executable and os.path.samefile(
#22 1140.0   File "/usr/local/lib/python3.8/genericpath.py", line 101, in samefile
#22 1140.0     s2 = os.stat(f2)
#22 1140.0 FileNotFoundError: [Errno 2] No such file or directory: '/usr/local/bin/pip'
#22 1140.0 Call stack:
#22 1140.0   File "/home/notebook-user/.local/bin/pip", line 8, in <module>
#22 1140.0     sys.exit(main())
#22 1140.0   File "/home/notebook-user/.local/lib/python3.8/site-packages/pip/_internal/cli/main.py", line 70, in main
#22 1140.0     return command.main(cmd_args)
#22 1140.0   File "/home/notebook-user/.local/lib/python3.8/site-packages/pip/_internal/cli/base_command.py", line 101, in main
#22 1140.0     return self._main(args)
#22 1140.0   File "/home/notebook-user/.local/lib/python3.8/site-packages/pip/_internal/cli/base_command.py", line 223, in _main
#22 1140.0     self.handle_pip_version_check(options)
#22 1140.0   File "/home/notebook-user/.local/lib/python3.8/site-packages/pip/_internal/cli/req_command.py", line 190, in handle_pip_version_check
#22 1140.0     pip_self_version_check(session, options)
#22 1140.0   File "/home/notebook-user/.local/lib/python3.8/site-packages/pip/_internal/self_outdated_check.py", line 236, in pip_self_version_check
#22 1140.0     logger.warning("[present-rich] %s", upgrade_prompt)
#22 1140.0   File "/usr/local/lib/python3.8/logging/init.py", line 1458, in warning
#22 1140.0     self._log(WARNING, msg, args, **kwargs)
#22 1140.0   File "/usr/local/lib/python3.8/logging/init.py", line 1589, in _log
#22 1140.0     self.handle(record)
#22 1140.0   File "/usr/local/lib/python3.8/logging/init.py", line 1599, in handle
#22 1140.0     self.callHandlers(record)
#22 1140.0   File "/usr/local/lib/python3.8/logging/init.py", line 1661, in callHandlers
#22 1140.0     hdlr.handle(record)
#22 1140.0   File "/usr/local/lib/python3.8/logging/init.py", line 954, in handle
#22 1140.0     self.emit(record)
#22 1140.0   File "/home/notebook-user/.local/lib/python3.8/site-packages/pip/_internal/utils/logging.py", line 179, in emit
#22 1140.0     self.handleError(record)
#22 1140.0 Message: '[present-rich] %s'
#22 1140.0 Arguments: (UpgradePrompt(old='22.2.1', new='23.0.1'),)```

**Comment by awalker4** [comment]
looks good!

@ryannikolaidis I saw this building the pipeline-general image as well. Seems to be harmless but we should probably note it for a second pass of these things.

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+USER ${NB_USER}
+WORKDIR ${HOME}
+RUN mkdir ${HOME}/.ssh && chmod go-rwx ${HOME}/.ssh \
+  &&  ssh-keyscan -t rsa github.com >> /home/${NB_USER}/.ssh/known_hosts
+
+ENV PYTHONPATH="${PYTHONPATH}:${HOME}"
+ENV PATH="/home/${NB_USER}/.local/bin:${PATH}"
+
+COPY unstructured unstructured
+COPY example-docs example-docs
```
Comment (ryannikolaidis) [inline Dockerfile:44]: How would you feel about moving this to after the pip install? Thinking: the install stage is painfully slow. If we move these after, then source or example-doc changes won't require re-installing every time (by benefit of caching)

**Comment by ryannikolaidis** [comment]
> looks good!
> 
> @ryannikolaidis I saw this building the pipeline-general image as well. Seems to be harmless but we should probably note it for a second pass of these things.

Ahh, okay, thanks! (wasn't sure if it was just my setup) Yea, totally agreed on both those points.

**Comment by cragwolfe** [comment]
On this error:
> #22 1140.0 FileNotFoundError: [Errno 2] No such file or directory: '/usr/local/bin/pip'

i'm curious what command is triggering that. But, since this already exists already:

    ln -s /usr/local/bin/python3.8 /usr/local/bin/python3

maybe add:

    ln -s /usr/local/bin/pip3.8 /usr/local/bin/pip3
    
or even

    ln -s /usr/local/bin/pip3.8 /usr/local/bin/pip
    
(as far as this image is concerned, we don't really care about any other pip version being pre-empted as the default)

**Review by awalker4 — APPROVED** [review]

**Pull request merged**